### PR TITLE
:bug: Source Recharge: migrated to 2021-11 api version

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/oncall/issues/2060
Due to changed in api version and incorrect pagination stream `subscription` read not all records

## How
Changed `next_page_token`: added `cursor` field, migrated to new api version 2021-11

### CODE REVIEW

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py [hunk 1]
```diff
@@ -33,24 +36,27 @@ def __init__(self, config, **kwargs):
...
     ) -> str:
         return self.name
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        stream_data = self.get_stream_data(response.json())
-        if len(stream_data) == self.limit:
-            self.page_num += 1
-            return {"page": self.page_num}
+        cursor = response.json().get("next_cursor")
+        if cursor:
```
Comment (atharva-satpute) [inline airbyte-integrations/connectors/source-recharge/source_recharge/api.py:51]:
```suggestion
        if cursor := response.json().get("next_cursor"):
```

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`fc84b43837`](https://github.com/airbytehq/airbyte/pull/27503/commits/fc84b438379c6435437d7a36c24a53b4bdc42754)) - ❌

⏲️  Total pipeline duration: 4331 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ❌        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5323849287)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`94a38cdc39`](https://github.com/airbytehq/airbyte/pull/27503/commits/94a38cdc39c09113ae57f1837b16bbffa85a97b7)) - ❌

⏲️  Total pipeline duration: 6746 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ❌        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5332032271)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`a497398a6a`](https://github.com/airbytehq/airbyte/pull/27503/commits/a497398a6a57664f630782ed8d46c64cbc698c97)) - ❌

⏲️  Total pipeline duration: 14537 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ❌        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5333215648)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`cc6ef09080`](https://github.com/airbytehq/airbyte/pull/27503/commits/cc6ef090806cae1d24aebcf9122213bf83a1966a)) - ✅

⏲️  Total pipeline duration: 458 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/daryna_source-recharge_oc206-missing-records/1687536189/cc6ef090806cae1d24aebcf9122213bf83a1966a/source-recharge/0.2.10/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

File: airbyte-integrations/connectors/source-recharge/source_recharge/api.py [hunk 2]
```diff
@@ -79,20 +84,6 @@ def should_retry(self, response: requests.Response) -> bool:
 
         return super().should_retry(response)
 
-    def stream_slices(
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-recharge/source_recharge/api.py:82]: what's the reason for removing slicing?

Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-recharge/source_recharge/api.py:82]: now connector uses `next_cursor` from api response to get next data, and first request uses start date from config or from stream state, so we don't need stream slices at all.

Comment (davydov-d) [reply airbyte-integrations/connectors/source-recharge/source_recharge/api.py:82]: Stream slicing enables us use checkpointing, therefore I suggest you leave stream slicing as is or modify if needed, but do not remove it. Use pagination inside the slice please.

**Review by davydov-d — CHANGES_REQUESTED** [review]

**Comment by darynaishchenko** [comment]
@davydov-d I can't leave stream slicing as it was, because it produces to duplicates. In stream with 2 records we receive 6. For pagination I'm using `next_page_token`. As I understand it's no way to retrieve cursor in `stream_slices` method. Only if we add cursor inside the record, but I don't think it's good idea.

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`f48acd3caf`](https://github.com/airbytehq/airbyte/pull/27503/commits/f48acd3caf89ae1c30a95f4eb360b9d716c705fc)) - ✅

⏲️  Total pipeline duration: 421 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/daryna_source-recharge_oc206-missing-records/1687776640/f48acd3caf89ae1c30a95f4eb360b9d716c705fc/source-recharge/0.2.10/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

**Review by davydov-d — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-recharge test report (commit [`89e3baa968`](https://github.com/airbytehq/airbyte/pull/27503/commits/89e3baa968841a8d5700822a9b64f0676235426f)) - ✅

⏲️  Total pipeline duration: 364 seconds

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-recharge/metadata.yaml | ✅        |
| Connector version semver check.                                        | ✅        |
| Connector version increment check.                                     | ✅        |
| QA checks                                                              | ✅        |
| Code format checks                                                     | ✅        |
| Connector package install                                              | ✅        |
| Build source-recharge docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                             | ✅        |
| Acceptance tests                                                       | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/daryna_source-recharge_oc206-missing-records/1687779521/89e3baa968841a8d5700822a9b64f0676235426f/source-recharge/0.2.10/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-recharge test
```

**Pull request merged**